SimulationAgent/Simulation/DeviceStatusLogic/Connect.cs

```diff
@@ -73,13 +73,14 @@ public void Run(object context)
                     {
                         this.log.Debug("Connect.Run calling this.devices.GetOrCreateAsync", () => new { this.deviceId, connectionTimeout.TotalMilliseconds });
 
-                        var task = this.devices.GetOrCreateAsync(this.deviceId);
+                        var task = this.devices.GetOrCreateAsync(this.deviceId, false);
                         task.Wait((int) connectionTimeout.TotalMilliseconds, actor.CancellationToken);
                         var device = task.Result;
 
                         this.log.Debug("Device credentials retrieved", () => new { this.deviceId });
 
                         actor.Client = this.devices.GetClient(device, this.protocol.Value, this.scriptInterpreter);
+                        actor.Client.ConnectAsync().Wait(connectionTimeout);
 
                         // Device Twin properties can be set only over MQTT, so we need a dedicated client
```

Comment:
Twin reads are supported over AMQP with the latest SDK - we should prep to upgrade.  Can you add a todo for it?

Reply:
👍  we need to test the perf and network reliability, will add task